contracts/BurnableStakeBank.sol

```diff
@@ -20,37 +20,40 @@ pragma solidity ^0.4.24;
 
 import 'openzeppelin-solidity/contracts/math/SafeMath.sol';
 import './interfaces/Lockable.sol';
-import './interfaces/IBurnableStakeBank.sol';
-import './interfaces/IBurnableERC20.sol';
+//import './interfaces/IBurnableStakeBank.sol';
+import './interfaces/BurnableERC20.sol';
+import './Co2knList.sol';
 
 
 // @title Contract for to keep track of stake (checkpoint history total staked at block) and burn tokens
-contract BurnableStakeBank is IBurnableStakeBank, Lockable {
+contract BurnableStakeBank is Lockable {
     using SafeMath for uint256;
 
     struct Checkpoint {
         uint256 at;
         uint256 amount;
     }
 
-    IBurnableERC20 public token;
+    Co2knList whitelist;
     Checkpoint[] public stakeHistory;
+    Checkpoint[] public burnHistory;
     uint256 public stakeLockBlockInterval = 1000;
 
     mapping (address => Checkpoint[]) public stakesFor;
+    mapping (address => Checkpoint[]) public burnsFor;
 
     // @param _token Token that can be staked.
-    constructor(IBurnableERC20 _token) public {
-        require(address(_token) != 0x0);
-        token = _token;
+    constructor(Co2knList _list) public {
+        require(address(_list) != 0x0);
+        whitelist = _list;
     }
 
     /**
      * @notice Stakes a certain amount of tokens.
      * @param amount Amount of tokens to stake.
      * @param data Data field used for signalling in more complex staking applications.
      */
-    function stake(uint256 amount, bytes data) public {
+    function stake(uint256 amount, bytes32 data) public {
```

Comment:
Will using bytes32 change the ABI / interface? I prefer bytes32, but the standard eip 900 specified bytes.  I don't want to break the interface compatibility in case staking wallets gets ported in the future.